astropy/cosmology/tests/test_connect.py

```diff
@@ -63,31 +83,39 @@ def test_readwrite_complete_info(self, cosmo, tmpdir, format):
         got = Cosmology.read(fname, format=format)
 
         assert got == cosmo
-        assert dict(got.meta) == dict(cosmo.meta)
+        if metaio:
+            assert dict(got.meta) == dict(cosmo.meta)
 
-    @pytest.mark.parametrize("format", readwrite_formats)
-    def test_readwrite_from_subclass_complete_info(self, cosmo_cls, cosmo, tmpdir, format):
+    @pytest.mark.parametrize("format, metaio, has_deps", readwrite_formats)
+    def test_readwrite_from_subclass_complete_info(
+            self, cosmo_cls, cosmo, tmpdir, format, metaio, has_deps):
         """
         Test writing from an instance and reading from that class, when there's
         full information saved.
         """
+        if not has_deps or not bs4Imported:
+            pytest.skip("missing a dependency", allow_module_level=True)
```

Comment:
I don't think we want to skip the module

```suggestion
            pytest.skip("missing a dependency")
```

Reply:
There are other instances that should be removed.

Reply:
Thanks for correcting.